cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -257,20 +248,77 @@ private String getSchemaCompatibility(
     }
   }
 
+  private Pair<String, String> getSubjectCompatibilityWithGlobalFallBack(
+      String environmentVal,
+      String topicName,
+      KafkaSupportedProtocol protocol,
+      String clusterIdentification) {
+    try {
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String compatibility =
+          getSubjectSchemaCompatibility(environmentVal, topicName, protocol, clusterIdentification);
+
+      if (!compatibility.equals(SCHEMA_COMPATIBILITY_NOT_SET)) {
+        return Pair.of(SUBJECT, compatibility);
+      }
+
+      String suffixUrl = environmentVal + "/config/";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol);
+
+      ResponseEntity<Map<String, String>> responseList =
+          getSubjectSchemaCompatibility(
+              reqDetails, new HashMap<>(), createSchmeaRegistryRequest(clusterIdentification));
+      log.info("Schema compatibility " + responseList);
+      return Pair.of(GLOBAL, responseList.getBody().get("compatibilityLevel"));
+    } catch (Exception e) {
+      log.error("Error in getting schema compatibility ", e);
+      return Pair.of(SCHEMA_COMPATIBILITY_NOT_SET, null);
+    }
+  }
+
+  private HttpEntity<Object> createSchmeaRegistryRequest(String clusterIdentification) {
+    HttpHeaders headers =
+        clusterApiUtils.createHeaders(clusterIdentification, KafkaClustersType.SCHEMA_REGISTRY);
+    HttpEntity<Object> request = new HttpEntity<>(headers);
+    return request;
+  }
+
+  private static ResponseEntity<Map<String, String>> getSubjectSchemaCompatibility(
```

Comment:
```suggestion
  private static ResponseEntity<Map<String, String>> invokeRestApiGet(
```

Or any other generic name? As it is not specific to schema compatibility I guess

Reply:
So this is used in just two places to only get the Schema compatibility at the subject/topic level so I don't really want to rename it to something more generic as I think its a good descriptive name of the method.

Reply:
There is actually two methods named the same but overloaded, Ill change the one that invokes the rest call to add "Request" at the end to make it explicit which is which.